packages/flutter/lib/src/rendering/selection.dart

```diff
@@ -0,0 +1,562 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:ui';
+
+import 'package:flutter/foundation.dart';
+import 'package:flutter/painting.dart';
+
+import 'box.dart';
+import 'layer.dart';
+import 'object.dart';
+
+/// The result after handling a selection event. This is used by the parent
```

Comment:
nit: first paragraph should just be one sentence (just add two '\n' after the ".").